### ISSUE TITLE
Issues with AppImage on Bazzite Deck OS

### ISSUE THREAD

**Issue by RodoMa92** [Spagett1/pineflash#74]
For whatever reason, pineflash will lock up using one single core completely each time I try to start up the AppImage file.

I've forked the project and compiled it myself and the UI now opens fine. The program though did not detect my iron until I've ran the program with sudo (might be a SeLinux permission problem tho). Another potential issues is that you assumes that the user already have dfu-util installed and if it's not it will not work for flashing the device (unless you provide already a binary build in the AppImage itself).

I'm installing it now to test if then this software will work for flashing my update.

Great job regardless for creating a native ui using Rust as a base language and not just use Electron again for it, my RAM appreciate the effort :)

Marco.

**Comment by RodoMa92** [comment]
Just tested the same on my archlinux desktop and it doesn't even try to start. Not sure why tho.

**Comment by Spagett1** [comment]
Hi, dfu-util ispackaged in the appimage with pineflash. Would you be able to run pineflash from the command line and send any output here?

Thanks

**Comment by RodoMa92** [comment]
> Hi, dfu-util ispackaged in the appimage with pineflash. Would you be able to run pineflash from the command line and send any output here?
> 
> Thanks

Yeah, it just hangs there until I kill it. No output anywhere.

**Comment by RodoMa92** [comment]
Strace log on my arch linux machine: 
[strace_log.txt](https://github.com/Spagett1/pineflash/files/14723814/strace_log.txt)

**Comment by Spagett1** [comment]
Interesting, in that case I'm not really sure what do do. You mention you are using bazzite, is that an immutable distro? When running the appimage could you ls the contents of /TMP and see if there is a folder names pineflash or similar?

Also could you confirm whether you have a v1 pinecil or v2?

As for the arch machine you can use the native package in the meanwhile. That should work. 

In regards to the Wayland issue that is strange and something I have encountered myself on my machine, a temporary solution is to unset the WAYLAND_DISPLAY variable which will cause it to fall back to xwayland, not ideal but as a temporary solution it works.

Unfortunately I have no experience with direct Wayland calls and I'm pretty unqualified to fix it if it is an upstream issue with the library I'm using. I can try updating the version used but that may take some time depending on how many breaking changes there are.

**Comment by Spagett1** [comment]
Interesting

**Comment by Spagett1** [comment]
Could you test the latest appimage from version 0.5.5?

**Comment by RodoMa92** [comment]
> Could you test the latest appimage from version 0.5.5?

Yup, on my KDE ArchLinux system it opened immediately with no fuss this time. It also detected it without running with sudo/anything else. Great job and thanks for reminding me by pinging, I already had forgotten that this problem happened in the first place :P

I'll test on Silverblue shortly and update this bug report shortly.

Thanks again for your work,

Marco.

**Comment by Spagett1** [comment]
Awesome, glad to hear.

Thanks for doing the test 👍

**Comment by razzeee** [comment]
Getting this on fedora

```
╰─ ./pineflash-0.5.5-x86_64.AppImage                                         ─╯
./pineflash-0.5.5-x86_64.AppImage: symbol lookup error: /tmp/.mount_pineflgbFCJh/usr/lib/libc.so.6: undefined symbol: __tunable_is_initialized, version GLIBC_PRIVATE

```

**Comment by hansmorb** [comment]
Similar issue on the latest 0.5.5 on Manjaro (Arch derivate):

`./pineflash-0.5.5-x86_64.AppImage 
Segmentation fault (core dumped)`

**Comment by Spagett1** [comment]
What is your glibc version (pacman -Qs glibc)
On manjaro I would recommend trying the arch package (.pkg.tar.zst)

**Comment by hansmorb** [comment]
> What is your glibc version (pacman -Qs glibc) On manjaro I would recommend trying the arch package (.pkg.tar.zst)

Thank you for the swift response!

`pacman -Qs glibc                                                 ✔ 
local/glibc 2.39-4
    GNU C Library
local/glibc-locales 2.39-4
    Pregenerated locales for GNU C Library
local/lib32-glibc 2.39-4
    GNU C Library (32-bit)
local/libnss_nis 3.2-1
    NSS NIS plugin for glibc
local/nss-mdns 0.15.1-1
    glibc plugin providing host name resolution via mDNS
`

I have installed the arch package successfully. I just wanted to use the AppImage since this is a tool I would just use once and do not want to keep having to (manually) update on my machine.
